### ISSUE TITLE
z3c.form fields with a default value cannot be set to None or ""

### ISSUE THREAD

**Issue by thet** [plone/Products.CMFPlone#1976]
I thought there is already an issue with that topic, but I didn't find one, so:

### What I did:

- I have a custom content type with a collection behavior. The query field of that behavior has a default value of a specific query string.
Emptying that query and saving it, saves an empty query, which is fine.
When revisiting the @@edit form, the default query is pre-filled. Saving the form again sets the default value instead, which is wrong.

- I have a custom content type with a text field and a default value. Emptying that value and saving clears the field, which is what I expect. When revisiting the @@edit form, the default query is pre-filled. Saving the form again sets the default value instead the empty value, which is wrong - same as above.

### This is related:

https://github.com/4teamwork/opengever.core/issues/1778

https://github.com/4teamwork/opengever.core/pull/2118

**Comment by hvelarde** [comment]
I'm experiencing the same issue on an add-on I was working it yesterday; the issue occurs on a `TextLine` field on the configlet:

https://github.com/collective/collective.newsticker/blob/master/src/collective/newsticker/controlpanel.py#L41-L48

**Comment by hvelarde** [comment]
I can confirm this doesn't happen with z3c.form 3.2.9.

**Comment by thet** [comment]
I have 3.2.11 and it's happening. Did you downgrade?

**Comment by thet** [comment]
I just tested with z3c.form 3.2.9 and z3c.formwidget.query 0.12 and I still have this issue, unfortunately...

**Comment by hvelarde** [comment]
yes, I downgraded and it started working fine; 3.2.9 is the version used in Plone 4.3.11.

**Comment by jensens** [comment]
Verified still present in z3c.form 6.0.1 (Plone 6). widget.py line 113: stored None/missing_value is reverted to field.default on re-edit. The code comment at line 108 acknowledges the issue. This is an upstream z3c.form bug that affects Plone.
